app/javascript/components/editor/TestsGroupedByTaskList.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useContext } from 'react'
+import { TestStatus, Test, AssignmentTask } from './types'
+import { GraphicalIcon } from '../common/GraphicalIcon'
+import { TestsGroup, TestWithToggle } from './TestsGroup'
+import { TestSummary } from './TestSummary'
+import { TasksContext } from '../Editor'
+
+type TaskWithTestsWithToggle = AssignmentTask & {
+  id: number
+  tests: TestWithToggle[]
+}
+
+const Tests = ({
+  tests,
+  language,
+}: {
+  tests: TestWithToggle[]
+  language: string
+}): JSX.Element => {
+  return (
+    <>
+      {tests.map((test) => (
+        <TestSummary
+          key={test.name}
+          test={test}
+          defaultOpen={test.defaultOpen}
+          language={language}
+        />
+      ))}
+    </>
+  )
+}
+
+const Title = ({ task }: { task: TaskWithTestsWithToggle }): JSX.Element => {
+  return (
+    <>
+      <span>Task {task.id}</span> <span>{task.title}</span>
+    </>
+  )
+}
+
+const JumpToInstructionButton = ({
+  taskId,
+}: {
+  taskId: number
+}): JSX.Element | null => {
+  const { switchToTask, showJumpToInstructionButton } = useContext(TasksContext)
+
+  return showJumpToInstructionButton ? (
+    <button type="button" onClick={() => switchToTask(taskId)}>
+      Jump to Instruction
+    </button>
+  ) : null
+}
+
+export function TestsGroupedByTaskList({
+  tests,
+  language,
+  tasks,
+}: {
+  tests: Test[]
+  language: string
+  tasks: AssignmentTask[]
+}): JSX.Element {
+  const testsWithIndex = tests.map((test, i) => ({ index: i + 1, ...test }))
+  const tasksWithTests = tasks.map((task, i) => ({
+    ...task,
+    id: i + 1,
+    tests: testsWithIndex.filter((test) => test.taskId === i + 1),
+  }))
+  const passed: TaskWithTestsWithToggle[] = tasksWithTests
+    .filter((task) =>
+      task.tests.every((test) => test.status === TestStatus.PASS)
+    )
+    .map((task) => ({
+      ...task,
+      tests: task.tests.map((test) => {
+        return { ...test, defaultOpen: false }
+      }),
+    }))
+
+  const failed: TaskWithTestsWithToggle[] = tasksWithTests
+    .filter((task) =>
+      task.tests.some(
+        (test) =>
+          test.status === TestStatus.FAIL || test.status === TestStatus.ERROR
+      )
+    )
+    .map((task, i) => ({
+      ...task,
+      tests: task.tests.map((test, j) => {
+        return {
+          ...test,
+          defaultOpen:
+            i === 0 &&
+            j ==
+              task.tests.findIndex(
+                (test) =>
+                  test.status === TestStatus.FAIL ||
+                  test.status === TestStatus.ERROR
+              ),
+        }
+      }),
+    }))
+
+  return (
+    <div className="tests-list">
```

Comment:
May want to make this an actual `ol`, cuz it is.

Reply:
I'm leaving the styling/markup to @iHiD :)